LayoutKitTests/CollectionTypeExtension.swift

```diff
@@ -7,16 +7,16 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 
 
-extension CollectionType {
+extension Collection {
     /// Returns the element at the specified index iff it is within bounds, otherwise nil.
     /// http://stackoverflow.com/questions/25329186/safe-bounds-checked-array-lookup-in-swift-through-optional-bindings
-    subscript (safe index: Index) -> Generator.Element? {
-        return indices.contains(index) ? self[index] : nil
+    subscript (safe index: Index) -> Iterator.Element? {
+        return distance(from: index, to: endIndex) < 1 ? self[index] : nil
```

Comment:
I tried this in a playground and it doesn't appear to work correctly.

I think it needs to look like this:
```swift
extension Collection where Index == Indices.Iterator.Element {
    subscript (safe index: Index) -> Iterator.Element? {
        return indices.contains(index) ? self[index] : nil
    }
}
```